src/mscorlib/shared/System/ReadOnlySpan.cs

```diff
@@ -321,5 +321,45 @@ public T[] ToArray()
         /// Returns a 0-length read-only span whose base is the null pointer.
         /// </summary>
         public static ReadOnlySpan<T> Empty => default(ReadOnlySpan<T>);
+
+        /// <summary>Gets an enumerator for this span.</summary>
+        public Enumerator GetEnumerator() => new Enumerator(this);
+
+        /// <summary>Enumerates the elements of a <see cref="ReadOnlySpan{T}"/>.</summary>
+        public ref struct Enumerator
+        {
+            /// <summary>The span being enumerated.</summary>
+            private readonly ReadOnlySpan<T> _span;
```

Comment:
Would it make sense to implement this in leaner way like array enumerator https://github.com/dotnet/coreclr/blob/8c74b3bcd691d842a891bc55909a18a8b1e53200/src/mscorlib/src/System/Array.cs#L2580 ?

Reply:
Yup.  Especially if/when we need to change Current to be ref returning, we need to do it that way.